### ISSUE TITLE
Special characters encoding issue on Mac

### ISSUE THREAD

**Issue by qi-wang-ocean** [AndreasMartin72/mksqlite#14]
Hi team,

First, thank you so much for your great work on mksqlite!! I'm seeking help on the special character encoding issue on my macOS Monterey v12.6. The special characters (e.g., degree °, micro μ) are not displaying properly. 
![Screen Shot 2022-09-14 at 9 11 26 AM](https://user-images.githubusercontent.com/92767057/190181492-e0d4a7df-b349-48a7-8adb-3dde8e93289c.png)
I set the default encoding of my Matlab as UTF-8, and the special characters look okay when using the {{sqlite}} matlab function provided by the database toolbox. And it has no problem on my colleague's PC. I'm using the latest version mksqlite downloaded here (https://github.com/a-ma72/mksqlite). I tried `mksqlite('convertUTF8',1);` but doesn't work to me.
I was just wondering if you happen to encounter something similar issues? Any advice or information would be really appreciated.

**Comment by a-ma72** [comment]
I'm sorry, I can't check the behavior because I don't have an Apple machine at my disposal. I was hoping that another user would answer this question in the meantime...

Best regards

**Comment by mirsev** [comment]
Hi, I have similar problem with degree sign. I have found that this symbol was saved with ASCII encoding (0xb0) in the database file, but SELECT statement returns it as a byte sequence [0xef 0xbf 0xbd] which is Unicode replacement character. So, I wrote a simple Matlab function to fix it:

```
function fixDegreeSymbol(tableName, column)
    hexstr = [ 'hex_' column ];
    s = mksqlite([ ...
        'SELECT *, hex(' column ') AS "' hexstr '" ' ...
        'FROM "' tableName '" WHERE hex(' column ') LIKE "%EFBFBD%"']);
    for k = 1:numel(s)
        fixedStr = char(hex2dec(reshape(strrep(s(k).(hexstr),'EFBFBD','B0'),2,[])'))';
        mksqlite(['UPDATE ' tableName ' SET ' column ' = ? WHERE rowid = ?'], ...
            fixedStr, s(k).rowid);
    end
end
```
Be careful, as this code assumes that `rowid` is used as the primary key in this table.
Also note that the Unicode replacement character can be substituted anywhere if some byte sequence in the text is meaningless for Unicode.

One interesting note: after fixing, the text in the database file is still in ASCII (with 0xb0 as the degree sign) but now SELECT statements return correct strings. I don't know why...

**UPDATE:** Later I have found that the same string is stored in 3 locations of the DB file and in 2 of them it is encoded as utf-8.
